src/renderer/components/Bonds/table/BondsTable.tsx

```diff
@@ -94,23 +94,22 @@ export const BondsTable: React.FC<Props> = ({ nodes, removeNode, network, goToNo
         columns={[nodeColumn, bondColumn, awardColumn, statusColumn, infoColumn, removeColumn]}
         dataSource={nodes.map((node) => ({ ...node, key: node.nodeAddress }))}
       />
-      {nodeToRemove && (
-        <ConfirmationModal
-          onClose={() => setNodeToRemove(null)}
-          onSuccess={() => {
-            removeNode(nodeToRemove)
-          }}
-          message={
-            <FormattedMessage
-              id="bonds.node.removeMessage"
-              defaultMessage="Are you sure you want to delete {node} node?"
-              values={{
-                node: <Styled.ConfirmationModalWalletText>{nodeToRemove}</Styled.ConfirmationModalWalletText>
-              }}
-            />
-          }
-        />
-      )}
+      <ConfirmationModal
+        visible={!!nodeToRemove}
+        onClose={() => setNodeToRemove(null)}
+        onSuccess={() => {
+          !!nodeToRemove && removeNode(nodeToRemove)
+        }}
+        message={
+          <FormattedMessage
+            id="bonds.node.removeMessage"
+            defaultMessage="Are you sure you want to delete {node} node?"
+            values={{
+              node: <Styled.ConfirmationModalWalletText>{nodeToRemove}</Styled.ConfirmationModalWalletText>
+            }}
+          />
+        }
+      />
```

Comment:
could you pls add such diff to your PR?
```diff
--- a/src/renderer/components/Bonds/table/BondsTable.tsx
+++ b/src/renderer/components/Bonds/table/BondsTable.tsx
@@ -2,6 +2,8 @@ import React, { useMemo, useState } from 'react'
 
 import { Address } from '@xchainjs/xchain-client'
 import { ColumnType } from 'antd/lib/table'
+import * as FP from 'fp-ts/function'
+import * as O from 'fp-ts/Option'
 import { FormattedMessage, useIntl } from 'react-intl'
 
 import { Network } from '../../../../shared/api/types'
@@ -74,19 +76,52 @@ export const BondsTable: React.FC<Props> = ({ nodes, removeNode, network, goToNo
     [goToNode]
   )
 
-  const [nodeToRemove, setNodeToRemove] = useState<Address | null>(null)
+  const [nodeToRemove, setNodeToRemove] = useState<O.Option<Address>>(O.none)
 
   const removeColumn: ColumnType<Node> = useMemo(
     () => ({
       key: 'remove',
       width: 40,
       title: '',
-      render: (_, { nodeAddress }) => <H.Delete deleteNode={() => setNodeToRemove(nodeAddress)} />,
+      render: (_, { nodeAddress }) => <H.Delete deleteNode={() => setNodeToRemove(O.some(nodeAddress))} />,
       align: 'right'
     }),
     [setNodeToRemove]
   )
 
+  const removeConfirmationProps = useMemo(() => {
+    const commonProps = {
+      onClose: () => setNodeToRemove(O.none),
+      onSuccess: FP.constVoid,
+      message: (
+        <FormattedMessage
+          id="bonds.node.removeMessage"
+          defaultMessage="Are you sure you want to delete {node} node?"
+          values={{
+            node: <Styled.ConfirmationModalWalletText>{nodeToRemove}</Styled.ConfirmationModalWalletText>
+          }}
+        />
+      )
+    }
+    return {
+      ...commonProps,
+      ...FP.pipe(
+        nodeToRemove,
+        O.fold(
+          () => ({
+            visible: false
+          }),
+          (node) => ({
+            visible: true,
+            onSuccess: () => {
+              removeNode(node)
+            }
+          })
+        )
+      )
+    }
+  }, [nodeToRemove])
+
   return (
     <>
       <Styled.Table
@@ -94,22 +129,7 @@ export const BondsTable: React.FC<Props> = ({ nodes, removeNode, network, goToNo
         columns={[nodeColumn, bondColumn, awardColumn, statusColumn, infoColumn, removeColumn]}
         dataSource={nodes.map((node) => ({ ...node, key: node.nodeAddress }))}
       />
-      <ConfirmationModal
-        visible={!!nodeToRemove}
-        onClose={() => setNodeToRemove(null)}
-        onSuccess={() => {
-          !!nodeToRemove && removeNode(nodeToRemove)
-        }}
-        message={
-          <FormattedMessage
-            id="bonds.node.removeMessage"
-            defaultMessage="Are you sure you want to delete {node} node?"
-            values={{
-              node: <Styled.ConfirmationModalWalletText>{nodeToRemove}</Styled.ConfirmationModalWalletText>
-            }}
-          />
-        }
-      />
+      <ConfirmationModal {...removeConfirmationProps} />
     </>
   )
 }

```

Reply:
Can do

Reply:
@thatStrangeGuyThorchain out of curiosity, could you explain why you suggest this change? Am curious about the tradeoffs you're considering

Reply:
thnx. just to avoid manual handling like `!!nodeToRemove` and have a strict separation between whether nodeToRemove exists or no